opentelemetry-sdk/src/resource/os.rs

```diff
@@ -1,46 +0,0 @@
-//! OS resource detector
-//!
-//! Detect the runtime operating system type.
-use crate::resource::ResourceDetector;
-use crate::Resource;
-use opentelemetry::KeyValue;
-use std::env::consts::OS;
-use std::time::Duration;
-
-/// Detect runtime operating system information.
-///
-/// This detector uses Rust's [`OS constant`] to detect the operating system type and
-/// maps the result to the supported value defined in [`OpenTelemetry spec`].
-///
-/// [`OS constant`]: https://doc.rust-lang.org/std/env/consts/constant.OS.html
-/// [`OpenTelemetry spec`]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/os.md
-#[derive(Debug)]
-pub struct OsResourceDetector;
```

Comment:
Actually would it make sense to mark this as deprecate for the next version and removed later?

Reply:
Sounds good to me, should I open a new PR and keep this one until next release?

Reply:
I think it is okay to move faster here - i.e remove right now. We only need to ensure that the alternate (the new crate) is published before the next release of sdk crate.